user/tests/wiring/sleep20/sleep20.cpp

```diff
@@ -661,5 +667,69 @@ test(26_System_Sleep_With_Configuration_Object_Ultra_Low_Power_Mode_Wakeup_By_Wi
     assertEqual(result.error(), SYSTEM_ERROR_NONE);
     assertEqual((int)result.wakeupReason(), (int)SystemSleepWakeupReason::BY_NETWORK);
 }
+#endif
 #endif // HAL_PLATFORM_WIFI
-#endif // HAL_PLATFORM_GEN == 3
+
+test(27_System_Sleep_With_Configuration_Object_Stop_Mode_Execution_Time) {
+    constexpr uint32_t SLEEP_DURATION_S = 3;
+    Serial.printf("    >> Device enters stop mode. Please reconnect serial after %ld s\r\n", SLEEP_DURATION_S);
+    Serial.println("    >> Press any key now");
+    while(Serial.available() <= 0);
+    while (Serial.available() > 0) {
+        (void)Serial.read();
+    }
+
+    Serial.println("    >> Connecting to the cloud");
+    Network.on();
+    Particle.connect();
+    waitUntil(Particle.connected);
+    Serial.println("    >> Connected to the cloud. You'll see the RGB is turned on after waking up.");
+
+    SystemSleepConfiguration config;
+    config.mode(SystemSleepMode::STOP)
+          .duration(SLEEP_DURATION_S * 1000);
+
+    time32_t enter = Time.now();
+    SystemSleepResult result = System.sleep(config);
+    time32_t exit = Time.now();
+
+    while (!Serial.isConnected());
+    // It might wait for up to 120s to turn off the modem
+    assertLessOrEqual(exit - enter, 120 + SLEEP_DURATION_S);
```

Comment:
Let's be a bit more strict and also add a check that the sleep was `>= SLEEP_DURATION_S`. Same in the next test.

Reply:
Updated.